shell.nix

```
@@ -86,6 +86,6 @@ mkShell {
   shellHook = ''
     # Prevent the avr-gcc wrapper from picking up host GCC flags
     # like -iframework, which is problematic on Darwin
-    unset NIX_TARGET_CFLAGS_COMPILE
+    unset NIX_CFLAGS_COMPILE_FOR_TARGET
```

If this was changed is it a good idea to unset both variables?

```suggestion
    unset NIX_CFLAGS_COMPILE_FOR_TARGET
    unset NIX_TARGET_CFLAGS_COMPILE
```

---

Maybe. I don't think it's *necessary* to unset both though, so I'm not sure what advantage there might be to doing so.

---

Backwards compatibility in case anyone is running a version that still uses `NIX_TARGET_CFLAGS_COMPILE`.

---

I guess it won't hurt anyway.

---

> Backwards compatibility in case anyone is running a version that still uses `NIX_TARGET_CFLAGS_COMPILE`.

Nixpkgs is pinned here, so nobody _can_ be using such a version.

---

Better in general to only have the code that is actually necessary, because now if somebody looks at this file, they will not be able to see which of those vars actually affects anything.

---

> Nixpkgs is pinned here, so nobody can be using such a version.

oh right, that makes sense. I've trimmed the last commit.